### PR TITLE
Fix browse by title filter doesn't work with diacritics

### LINKED ISSUE #8434: Browse by title filter doesn't work with diacritics
**Describe the bug**
In DSpace 7.3 the behavior of startsWith param of the browse by title index has been change to work as a filter instead of "Jump to value". This change makes more evident a problem when filtering accented titles, when an user searches for titles that contains diacritics some titles are not returned.

**To Reproduce**
1. Publish an item with an accented title and another with an unaccented title. By example: "Número 1" and "Numero 2"
2. Filter by title with value: "Número". No result returned. 
3. Filter by title with value: "Numero". The unaccented title is returned but not the accented one. 

**Expected behavior**
Ideally, the accents should be ignored and both searches should return the two items.

**Related work**
The same problem for the subject and author indexes has been addressed here:
Issue: https://github.com/DSpace/DSpace/issues/7381
PR: https://github.com/DSpace/DSpace/pull/2276

### PR DESCRIPTION
**Pull request by toniprieto**
## References

* Fixes #8434

## Description

Change the default normalization of the indexed text used for sorting and filtering titles to strip the diacritics.

It is similar to the PR applied for filtering accented content in subject/author indexes: #2698

## Instructions for Reviewers

To test the changes:

* Publish an item with an accented title and another with an unaccented title. By example: "Número 1" and "Numero 2"
* Filter by title with value: "Número". Should return the two items.
* Filter by title with value: "Numero".  Should return the two items.

List of changes in this PR:
* Apply the StripDiacritics filter to the normalization of indexed titles
* Applies the same normalization to the user-input text when the user is filtering by title
* Include a new test to check the behavior with diacritics
* Correct two comments that I forgot to change in PR #2698

## Checklist
_This checklist provides a reminder of what we are going to look for when reviewing your PR. You need not complete this checklist prior to creating your PR (draft PRs are always welcome). If you are unsure about an item in the checklist, don't hesitate to ask. We're here to help!_

- [x] My PR is small in size (e.g. less than 1,000 lines of code, not including comments & integration tests). Exceptions may be made if previously agreed upon.
- [x] My PR passes Checkstyle validation based on the [Code Style Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Style+Guide).
- [x] My PR includes Javadoc for _all new (or modified) public methods and classes_. It also includes Javadoc for large or complex private methods.
- [x] My PR passes all tests and includes new/updated Unit or Integration Tests based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [x] If my PR includes new, third-party dependencies (in any `pom.xml`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/main/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.
- [x] If my PR modifies the REST API, I've linked to the REST Contract page (or open PR) related to this change.

### CODE REVIEW

**Review by corrad82-4s — CHANGES_REQUESTED** [review]
Overall this code looks good to me, I have just reported a Minor (possible) issue.

File: dspace-api/src/main/java/org/dspace/browse/BrowseEngine.java [hunk 1]
```diff
@@ -203,7 +203,12 @@ private BrowseInfo browseByItem(BrowserScope bs)
             // get the table name that we are going to be getting our data from
             dao.setTable(browseIndex.getTableName());
 
-            dao.setStartsWith(StringUtils.lowerCase(scope.getStartsWith()));
+            if (scope.getBrowseIndex().getDataType().equals(OrderFormat.TITLE)) {
```
[the remark this answers is not in the document]
Comment (toniprieto) [reply dspace-api/src/main/java/org/dspace/browse/BrowseEngine.java]: Change added to prevent NPEs. Thanks!

**Comment by corrad82-4s** [comment]
@tdonohue , @toniprieto  after last change, code looks good to me.

**Review by corrad82-4s — APPROVED** [review]

**Review by tdonohue — APPROVED** [review]
👍 Thanks @toniprieto ! This looks correct to me and the ITs prove that it works as expected.

**Pull request merged**